src/jit/importer.cpp

```diff
@@ -6830,7 +6830,7 @@ var_types  Compiler::impImportCall(OPCODE                  opcode,
     //-------------------------------------------------------------------------
     // The "this" pointer
 
-    if (!(mflags & CORINFO_FLG_STATIC) || opcode == CEE_NEWOBJ)
```

Comment:
Note that `opcode == CEE_NEWOBJ` condition was unnecessary in the original code because of `CORINFO_FLG_STATIC` is never set for instance constructors.